pkgs/os-specific/linux/plymouth/patch/0001-Change-default-screen-orientation-to-COUNTER_CLOCKWI.patch

```diff
@@ -0,0 +1,26 @@
+From 6b11c25e6c3a166668f78eef67b76a97fb865f4d Mon Sep 17 00:00:00 2001
+From: "Mathieu A.-Tetreault" <alexandretm@amotus.ca>
+Date: Fri, 18 Oct 2019 10:30:11 -0400
+Subject: [PATCH] Change default screen orientation to COUNTER_CLOCKWISE(Left
+ Side Up)
+
+---
+ src/plugins/renderers/drm/plugin.c | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/src/plugins/renderers/drm/plugin.c b/src/plugins/renderers/drm/plugin.c
+index 1ff2b29..9d57899 100644
+--- a/src/plugins/renderers/drm/plugin.c
++++ b/src/plugins/renderers/drm/plugin.c
+@@ -438,7 +438,7 @@ ply_renderer_connector_get_rotation_and_tiled (ply_renderer_backend_t      *back
+         drmModePropertyPtr prop;
+         int i;
+ 
+-        *rotation = PLY_PIXEL_BUFFER_ROTATE_UPRIGHT;
++        *rotation = PLY_PIXEL_BUFFER_ROTATE_COUNTER_CLOCKWISE;
+         *tiled = false;
+ 
+         for (i = 0; i < connector->count_props; i++) {
+-- 
```

Comment:
A little bit below, there seems to be a `connector_orientation_prop_to_rotation` fn used which extract an enum from a `drmModePropertyPtr`. This enum seems to have possible values: {"Upside Down", "Left Side Up", "Right Side Up"} and convert those to {PLY_PIXEL_BUFFER_ROTATE_UPSIDE_DOWN, PLY_PIXEL_BUFFER_ROTATE_COUNTER_CLOCKWISE, PLY_PIXEL_BUFFER_ROTATE_CLOCKWISE, PLY_PIXEL_BUFFER_ROTATE_UPRIGHT }. 

This drm mode property is obtained using the `drmModeGetProperty (backend->device_fd, connector->props[i])` expression. Did you investigate if it is possible to set the drm mode property on the device instead? If so, it would allow us to avoid patching the code is simply configure a device.

Reply:
[Kernel Mode Setting (KMS) — The Linux Kernel documentation](https://www.kernel.org/doc/html/v4.15/gpu/drm-kms.html)

Reply:
Used `drmModeGetProperty` accessor is defined in driver [xf86drmMode.h source code [include/xf86drmMode.h] - Woboq Code Browser](https://code.woboq.org/qt5/include/xf86drmMode.h.html).

Reply:
Some discussion about the `"panel orientation"` kernel mode property:

[linux: Screen rotation issue fully fixed now... : GPDPocket](https://www.reddit.com/r/GPDPocket/comments/7s0wn1/linux_screen_rotation_issue_fully_fixed_now/)

Related:

[ubuntu - set kms specific resolution at boot time - Unix & Linux Stack Exchange](https://unix.stackexchange.com/questions/520789/set-kms-specific-resolution-at-boot-time)

Reply:
[Plymouth - ArchWiki](https://wiki.archlinux.org/index.php/plymouth)

 > Plymouth primarily uses KMS (Kernel Mode Setting) to display graphics. In EFI/UEFI systems, plymouth can utilize the EFI framebuffer. If you can't use KMS in e.g. because you are using a proprietary driver, or if you don't want to use EFI framebuffer, consider using Uvesafb as it works with widescreen resolutions.

> If you have neither KMS nor a framebuffer, Plymouth will fall back to text-mode. 

What I understand is that if we wanted to start plymouth earlier in the boot process we would use the EFI framebuffer instead of the KMS backend.

Reply:
### Some mode settings refs

[The DRM/KMS subsystem from a newbie's point of view - brezillon-drm-kms.pdf](https://events.static.linuxfound.org/sites/events/files/slides/brezillon-drm-kms.pdf)

[Mode setting - Wikipedia](https://en.wikipedia.org/wiki/Mode_setting)

[Direct Rendering Manager - Wikipedia](https://en.wikipedia.org/wiki/Direct_Rendering_Manager)

[linux/drm-kms.rst at master · torvalds/linux](https://github.com/torvalds/linux/blob/master/Documentation/gpu/drm-kms.rst)

[Xorg/Guide - Gentoo Wiki](https://wiki.gentoo.org/wiki/Xorg/Guide)

 >  KMS conflicts with legacy framebuffer drivers, which must remain disabled in the kernel configuration.

Might explain why the fb kernel wasn't effective.

[ubuntu - set kms specific resolution at boot time - Unix & Linux Stack Exchange](https://unix.stackexchange.com/questions/520789/set-kms-specific-resolution-at-boot-time)

[Nvidia and high resolution console with KMS? : archlinux](https://www.reddit.com/r/archlinux/comments/6x8mzv/nvidia_and_high_resolution_console_with_kms/)

[kernel - How to set a video mode in linux with kms/drm? - Unix & Linux Stack Exchange](https://unix.stackexchange.com/questions/10634/how-to-set-a-video-mode-in-linux-with-kms-drm)



There seems to be a grub way of setting modes. There might be some way to do the same through systemd boot.


[Nvidia and high resolution console with KMS? : archlinux](https://www.reddit.com/r/archlinux/comments/6x8mzv/nvidia_and_high_resolution_console_with_kms/)

Reply:
### Some systemd-boot config refs

[systemd-boot - ArchWiki](https://wiki.archlinux.org/index.php/systemd-boot)

[loader.conf(5) — Arch manual pages](https://jlk.fjfi.cvut.cz/arch/manpages/man/loader.conf.5)

There does not seem to be a lot of possible configurations with systemd-boot outside of kernel args and console mode.

Reply:
### Some linux kernel args refs

[linux/kernel-parameters.txt at master · torvalds/linux](https://github.com/torvalds/linux/blob/master/Documentation/admin-guide/kernel-parameters.txt)

Reply:
### Kernel Module configuration ref 

[Kernel module - ArchWiki](https://wiki.archlinux.org/index.php/Kernel_module)


 >  Using kernel command line
 >  
 >  If the module is built into the kernel, you can also pass options to the module using the kernel command line. For all common bootloaders, the following syntax is correct:
 >  
 >  module_name.parameter_name=parameter_value
 >  
 >  For example:
 >  
 >  thinkpad_acpi.fan_control=1
 >  
 >  Simply add this to your bootloader's kernel-line, as described in Kernel Parameters. 

This would seem to be the way to go to configuration the video driver.

Reply:
> A little bit below, there seems to be a `connector_orientation_prop_to_rotation` fn used which extract an enum from a `drmModePropertyPtr`. This enum seems to have possible values: {"Upside Down", "Left Side Up", "Right Side Up"} and convert those to {PLY_PIXEL_BUFFER_ROTATE_UPSIDE_DOWN, PLY_PIXEL_BUFFER_ROTATE_COUNTER_CLOCKWISE, PLY_PIXEL_BUFFER_ROTATE_CLOCKWISE, PLY_PIXEL_BUFFER_ROTATE_UPRIGHT }.
> 
> This drm mode property is obtained using the `drmModeGetProperty (backend->device_fd, connector->props[i])` expression. Did you investigate if it is possible to set the drm mode property on the device instead? If so, it would allow us to avoid patching the code is simply configure a device.

As per my understanding, the drmModGetProperty function, reads the information from the graphic cards/monitor. It relies on the fact that some display/devices have accelerometers within them to return the orientation. Our devices don't have an accelerometer.

Reply:
Yes, this EDID option might be worth investigating too.